app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala

```diff
@@ -299,79 +299,82 @@ object BitcoindRpcBackendUtil extends Logging {
       bitcoind: BitcoindRpcClient,
       interval: FiniteDuration = 10.seconds)(implicit
       system: ActorSystem,
-      ec: ExecutionContext): Future[Cancellable] = {
-    val walletSyncStateF = wallet.getSyncState()
-    val resultF: Future[Cancellable] = for {
-      walletSyncState <- walletSyncStateF
+      ec: ExecutionContext): Future[Unit] = {
+
+    for {
+      walletSyncState <- wallet.getSyncState()
+      rescanning <- wallet.isRescanning()
     } yield {
-      val numParallelism = Runtime.getRuntime.availableProcessors()
-      val atomicPrevCount: AtomicInteger = new AtomicInteger(
-        walletSyncState.height)
-      val processing = new AtomicBoolean(false)
-      system.scheduler.scheduleWithFixedDelay(0.seconds, interval) { () =>
-        {
-          if (processing.compareAndSet(false, true)) {
-            logger.trace("Polling bitcoind for block count")
-            val res = bitcoind.getBlockCount.flatMap { count =>
-              val prevCount = atomicPrevCount.get()
-              if (prevCount < count) {
-                logger.info(
-                  s"Bitcoind has new block(s), requesting... ${count - prevCount} blocks")
+      if (!rescanning) {
```

Comment:
This means if we start the wallet and we are rescanning, we will never start polling bitcoind after the rescan is finished.